File: Games/BombermanGame.Shared/Interfaces/IPlayer.cs
using Games.BombermanGame.Shared.Enums;

namespace Games.BombermanGame.Shared.Interfaces
{
    public interface IPlayer
    {
        /// <summary>
        /// Short name for AI player
        /// </summary>
        string Nickname { get; }
        /// <summary>
        /// AI strategy description for tournament information
        /// </summary>
        string StrategyDescription { get; }

        /// <summary>
        /// Is bot in debug mode (disabled timeout for turn) 
        /// </summary>
        bool IsDebugMode { get; }
        
        /// <summary>
        /// The game name for which Ai/bot was developed
        /// </summary>
        string AiDevelopedForGame { get; }

        /// <summary>
        /// Method which is used for turn calculation
        /// </summary>
        /// <param name="gameInfo">Whole game information that can be required</param>
        /// <param name="currentPlayerInfo">Info about current player</param>
        /// <returns>Enum that describe current player turn request</returns>
        PlayerTurnEnum Turn(IGameInfo gameInfo, IPlayerInfo currentPlayerInfo);

        void OnTurnTimeExceeded();
    }
}

File: BombermanClient_C_PlusPlus/Games/BombermanGame.Shared/Interfaces/IPlayer.cpp
//
// Created by Andrey on 02.12.2023.
//

#include "IPlayer.h"

namespace Games_BombermanGame_Shared_Interfaces {
    const string IPlayer::GetNickname() const {
        return this->_nickname;
    }

    const string IPlayer::GetStrategyDescription() const {
        return _strategyDescription;
    }

    bool IPlayer::GetIsDebugMode() const {
        return _isDebugMode;
    }

    const string IPlayer::GetAiDevelopedForGame() const {
        return _aiDevelopedForGame;
    }

    IPlayer::IPlayer(char* nickname,
                     char* strategyDescription,
                     bool isDebugMode,
                     char* aiDevelopedForGame) :
            _nickname(nickname),
            _strategyDescription(strategyDescription),
            _isDebugMode(isDebugMode),
            _aiDevelopedForGame(aiDevelopedForGame) {}
} // Games_BombermanGame_Shared_Interfaces